### PR TITLE
add package mongodb

### PR DESCRIPTION
**Pull request by MedzikUser**
#7835

### CODE REVIEW

**Comment by MedzikUser** [comment]
@xeffyr @thunder-coding @landfillbaby any ideas? pip should be installed at least it is in setup-ubuntu.sh

**Comment by thunder-coding** [comment]
https://github.com/termux/termux-packages/blob/master/scripts/setup-ubuntu.sh#L61

setup-ubuntu.sh already installs pip, consider using `pip3` directly instead of `python3.9 -m pip`

**Comment by MedzikUser** [comment]
> https://github.com/termux/termux-packages/blob/master/scripts/setup-ubuntu.sh#L61
> 
> setup-ubuntu.sh already installs pip, consider using `pip3` directly instead of `python3.9 -m pip`

not working

```
Traceback (most recent call last):
  File "/usr/bin/pip3", line 5, in <module>
    from pip._internal.cli.main import main
ModuleNotFoundError: No module named 'pip'
```

**Comment by tomty89** [comment]
I have no idea why you are seeing the error. However, I don't think the right way for this is using pip in the build script. Rather you should add what's necessary (including scons) to the setup-* scripts, I think.

I wonder if scons works with our build system / the NDK btw. Either way, we don't have `MAKEFLAGS` here. You can't just copy what's in the PKBGUILD from AUR.

Finally,
https://github.com/termux/termux-packages/blob/9d020f82ec10f83c722c8ca6db2f3b67ff5f6714/scripts/properties.sh#L28

**Comment by thunder-coding** [comment]
@MedzikUser, MongoDB is being built for host, according to the previous CI run which failed on installation step. Notice `gcc` and `g++` instead of `aarch64-android-linux-clang` or other respective cross compiler

**Comment by MedzikUser** [comment]
> @MedzikUser, MongoDB is being built for host and not for Android, according to the previous CI run which failed on installation step. Notice `gcc` and `g++` instead of `aarch64-android-linux-clang` or other respective cross compiler

Well I was just wondering why there was g++

**Comment by thunder-coding** [comment]
This seems to be the same thing I faced while building V8. The build files by MongoDB doesn't respect CC, CXX and other environment variables

**Comment by MedzikUser** [comment]
on stackoverflow it was written that env files (CC, CXX) are set in some file

**Comment by tomty89** [comment]
Please familiarize yourself with the build system first. `TERMUX_PKG_EXTRA_MAKE_ARGS` is not what you think it is. Besides, if you aren't even sure how to resolve the pip install into necessary addition for the "setup" script(s), I'm not sure if you should (blindly) attempt further.

**Comment by thunder-coding** [comment]
> Add CC=clang70 CXX=clang++70 to the scons options, when building

MongoDB docs specify passing CC and CXX to scons options for building for FreeBSD, probably the same needs to be done here.

**Comment by MedzikUser** [comment]
> Please familiarize yourself with the build system first. `TERMUX_PKG_EXTRA_MAKE_ARGS` is not what you think it is. Besides, if you aren't even sure how to resolve the pip install into necessary addition for the "setup" script(s), I'm not sure if you should (blindly) attempt further.

I already fixed pip

**Comment by tomty89** [comment]
Resolve != solve / fix. Besides, the extra apt calls are probably unnecessary / inappropriate / unresolved.

**Comment by MedzikUser** [comment]
> Resolve != solve / fix. Besides, the extra apt calls are probably unnecessary / inappropriate / unresolved.

see https://github.com/termux/termux-packages/runs/4060224902?check_suite_focus=true#step:3:1153

File: packages/mongodb/build.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+termux_step_make() {
+	cd $TERMUX_PKG_SRCDIR
+	echo "CC: $CC, CXX: $CXX"
+
+	sed -i '/"-ggdb" if not env.TargetOSIs/d' SConstruct
+
+	pip3 install -r etc/pip/compile-requirements.txt
+
+	export SCONSFLAGS="$TERMUX_PKG_EXTRA_MAKE_ARGS"
+	python3 buildscripts/scons.py install-mongod CC=aarch64-android-linux-clang CXX=aarch64-android-linux-clang++ MONGO_VERSION="$TERMUX_PKG_VERSION" --disable-warnings-as-errors
```
Comment (thunder-coding) [inline packages/mongodb/build.sh:20]: Do not hardcode this. Better use `CC=$CC CXX=$CXX`

**Review by thunder-coding — CHANGES_REQUESTED** [review]

File: packages/mongodb/build.sh (hunk 1 above)
Comment (MedzikUser) [reply packages/mongodb/build.sh:20]: That's what I thought, I also wanted to see what CC and CXX is so I added an echo

**Comment by thunder-coding** [comment]
> I'm not sure if you should (blindly) attempt further.

I feel @tomty89 is correct, you need to have basic knowledge about the current build system first. Let some other contributor work on MongoDB. Also your commits, they need to be squashed, and instead of merging master branch, rebase against it.

**Comment by MedzikUser** [comment]
```
Checking if C++ compiler "aarch64-linux-android-clang++" is GCC... no
2021-11-01T09:32:23.8655298Z Checking if C++ compiler "aarch64-linux-android-clang++" is clang... no
2021-11-01T09:32:23.8656473Z Couldn't identify the C++ compiler
```

**Pull request closed without merging**